### DISCUSSION TITLE
How to append to shared_dirs?

### DISCUSSION THREAD

**Discussion by mikestreety** [sourcebroker/deployer-extended-typo3#30] [Q&A] [answered]
Hey,

Because, in the TYPO3 11 config file the `shared_dirs` config is set [using a function](https://github.com/sourcebroker/deployer-extended-typo3/blob/master/deployer/11/set.php#L9) I can't use the `add('shared_dirs', [])` function in my local config.

Is there a way I can add to the shared dirs, rather than redeclaring them all (so I can upgrade TYPO3 version without worrying about keeping it in sync)?

Thanks

**Comment by kszymukowicz** [comment, marked as answer]
hi, I would use

```
set('shared_dirs', [...call_user_func(get('shared_dirs')), 'newDir']);
```

**Comment by mikestreety** [comment]
Thanks - worked perfectly

**Comment by mikestreety** [comment]
**Update.**

With version 20 (and Deployer version 7), this is now done with:

```php
set('shared_dirs', [
	...get('shared_dirs'),
	'newDir',
]);
```
